Yosemite/Yosemite/Stores/CardPresentPaymentStore.swift

```diff
@@ -392,48 +390,32 @@ private extension CardReaderConfigError {
 
 // MARK: Networking Methods
 private extension CardPresentPaymentStore {
-    /// Switch the store to use WCPay as the backend.
-    /// Does nothing if the store is already using WCPay.
-    /// WCPay is the initial default for the store.
+    /// Sets the store to use a given payment gateway
     ///
-    func useWCPayAsBackend() {
-        guard usingBackend != .wcpay else {
-            return
-        }
-
-        usingBackend = .wcpay
-    }
-
-    /// Switch the store to use Stripe as the backend.
-    /// Does nothing if the store is already using Stripe.
-    ///
-    func useStripeAsBackend() {
+    func use(paymentGatewayAccount: PaymentGatewayAccount) {
         guard allowStripeIPP else {
             DDLogError("useStripeAsBackend called when stripeExtensionInPersonPayments disabled")
             return
         }
 
-        guard usingBackend != .stripe else {
+        guard paymentGatewayAccount.isWCPay else {
```

Comment:
If, as we mentioned in another PR, we extracted a class to track the actual gateway, we would just need to pass this parameter through to that class, and the rest of the methods would just need to query that class for the "activeGateway"